bpf/lib/common.h

```diff
@@ -78,7 +78,6 @@ enum {
 
 #define CILIUM_CALL_DROP_NOTIFY			1
 #define CILIUM_CALL_ERROR_NOTIFY		2
-#define CILIUM_CALL_SEND_ICMP6_ECHO_REPLY	3
```

Comment:
Not sure if we just leave a gap here, or change all the macro definitions below.

Reply:
I'm unsure if that can cause issues on upgrade. Maybe @ti-mo knows?

Reply:
Thanks for the ping! Unfortunately, I think we may need to renumber to cause a change in CILIUM_CALL_SIZE, which will force map migration to recreate the prog array from scratch. That is, CILIUM_CALL_SIZE needs to differ from whatever 1.13 version is used during the upgrade test. v1.13.2 has `CILIUM_CALL_SIZE = 40`, so map migration will still take place if we drop it back down to 42 on main.

#20691 is relevant here, we should be able to get that done over the coming weeks. This should remove another potential cause of missed tail calls and undefined behaviour during minor upgrades when tail call maps are currently being re-used.

Reply:
Thank you for your information @ti-mo , that was very helpful. Just one thing to confirm, does changing `CILIUM_CALL_SIZE` break the upgrade test? The [test suite](https://jenkins.cilium.io/job/Cilium-PR-K8s-1.25-kernel-5.4/24/testReport/junit/Suite-k8s-1/25/K8sUpdates_Tests_upgrade_and_downgrade_from_a_Cilium_stable_image_to_master/) `Tests upgrade and downgrade from a Cilium stable image to master` failed because migrate-svc had restarted unexpectedly, do you have any clue where I can look into?

Reply:
To pass upgrade test suite, I leave a gap here with additional comments to explain what happened here lol

Reply:
Does it fail on every run with the change to `CILIUM_CALL_SIZE` included? The run you linked seems to fail on downgrade. Either way, the prog array gets recreated when switching between main/master and 1.13, so there shouldn't be any missed tail calls.

Looking at the `cilium-metrics-list.md` files in the sysdump seems to corroborate this, there are no `cilium_drop_` entries with `reason=Missed tail call` (these are normally echoed in the test stdout), but there are other drop reasons that might be of interest in finding the cause.

You may be seeing a non-tail call-related flake that may or may not be caused by this PR.

Reply:
Thanks for your help Timo, I ran K8sUpdates locally with new `CILIUM_CALL_SIZE` for 3 times and I'm afraid K8sUpdates fails on every run LOL